gdal/ogr/ogrsf_frmts/oci/ogrocisession.cpp

```diff
@@ -130,9 +130,8 @@ int OGROCISession::EstablishSession( const char *pszUseridIn,
 
     ub4 eCred = OCI_CRED_RDBMS;
 
-    if( EQUAL(pszDatabaseIn, "") &&
-        EQUAL(pszPasswordIn, "") &&
-        EQUAL(pszUseridIn, "/") )
+    if( EQUAL(pszPasswordIn, "") &&
```

Comment:
The EQUAL(pszUseridIn, "/") test has been changed to EQUAL(pszUseridIn, "").  Cannot pszUseridIn == '/' happy anymore ? Or what will happen in that case ?
Sorry if my question is a bit dump. I'm not familiar with Oracle

Reply:
I am not an expert on this either. From the documentation (https://docs.oracle.com/en/database/oracle/oracle-database/21/lnoci/connect-authorize-and-initialize-functions.html#GUID-D60E78B5-E600-4CDE-87A7-B4ACDA9FEE5E)
"OCI_CRED_EXT - Authenticate using external credentials. No user name or password is provided. "
"The second method is to use external credentials. No attributes need to be set on the user session handle before calling OCISessionBegin(). The credential type is OCI_CRED_EXT. This is equivalent to the Oracle7 'connect /' syntax. If values have been set for OCI_ATTR_USERNAME and OCI_ATTR_PASSWORD, then these are ignored if OCI_CRED_EXT is used. "

From ogrocidatasource.cpp, pszUseridIn is set "/" only when the user, pwd and database are all empty. It seems that that is the old way (Oracle 7?) of specifying the credentials. If we want backward compatibility,  I can keep the original code and add the new way, like this: 
    if( EQUAL(pszDatabaseIn, "") &&
        EQUAL(pszPasswordIn, "") &&
        EQUAL(pszUseridIn, "/")  ||
        EQUAL(pszPasswordIn, "") && 
        EQUAL(pszUseridIn, ""))

Reply:
> I am not an expert on this either

Hum... I can't really tell on what's the appropriate change, but I don't think we need to support ancient/deprecated Oracle. If supports for the "traditional" authentication support that was handled by the driver up to know still works with recent enough versions of Oracle, in addition to the "wallet" one, that seems fine to me.
Did you try running the autotest/gdrivers/georaster.py tests ? No idea if/how they work.
(It's a bit of a pain to support that proprietary stuff since we can't easily tests that in continuous integration)

Reply:
I didn't run the georaster.py test because it won't test the new feature which requires the oracle wallet setup. I only manually tested the driver to make sure the connection like "geor:/@orcl" works. The current change should not affect the "traditional" connect string where user and pwd are not empty.

Reply:
> The current change should not affect the "traditional" connect string where user and pwd are not empty.

sounds good